### ISSUE TITLE
Cannot read property 'key.filepath' of undefined

### ISSUE THREAD

**Issue by LinusU** [vknabel/vscode-swift-development-environment#10]
```text
[Error - 12:29:44] Request textDocument/definition failed.
  Message: Request textDocument/definition failed with message: Cannot read property 'key.filepath' of undefined
  Code: -32603 
```

I'm not exactly sure what triggered this, but it happened while trying to autocomplete `internal let valet: Val...` in the code below:

```swift
import Foundation

import PromiseKit
import Valet

open class BioPass {
    enum Error: Swift.Error {
        case notAvailable
        case failedEncodingPassword
        case failedCreatingSAC
        case unhandledKeychainError(OSStatus)
        case unexpectedPasswordData
    }

    let serviceName: String

    internal let valet: Val

    public init(_ serviceName: String = Bundle.main.bundleIdentifier!) {

        if serviceName == "" { fatalError("serviceName cannot be an empty string") }

        self.serviceName = serviceName
    }
```

Both `PromiseKit` and `Valet` are installed as dependencies with Swift Package Manager.

**Comment by vknabel** [comment]
@LinusU probably your autocompletion did stop working and did work after restarting vscode again?

**Comment by LinusU** [comment]
I also got this error:

```text
[Error - 19:17:42] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot read property 'key.kind' of undefined
  Code: -32603 
```

> probably your autocompletion did stop working and did work after restarting vscode again?

Yes, I tried restarting it now and it seems to be working again 😄 (yeah I have been working for 24h without code completion 😆 😱)

**Comment by vknabel** [comment]
@LinusU in that case I found the where the error starts manifesting. Although I don‘t know its exact reason yet. Either it’s the classic „SourceKit has been terminated“-error as known by Xcode or it’s a race condition within this extension. 

This produces a bunch of slightly different error messages. But at least there is a temporary workaround (restarting). 

In general if this extension behaves strange, saving the document, adding whitespace and saving it again may help you. Thereafter you always can try a restart. 

I wish the language support for Swift for vscode would be as good as for typescript, so keep the issues coming.

**Comment by vknabel** [comment]
@LinusU can you have a look on the following build? Unfortunately the error didn't occur since then.
This build additionally contains hover documentation as proposed in #11.

[vscode-swift-development-environment-2.3.0.vsix.zip](https://github.com/vknabel/vscode-swift-development-environment/files/1821516/vscode-swift-development-environment-2.3.0.vsix.zip)

**Comment by LinusU** [comment]
Downloading and trying out now 👌

**Comment by LinusU** [comment]
Hmm, seems like I got a very similar error:

```text
[Error - 20:16:38] Request textDocument/hover failed.
  Message: Request textDocument/hover failed with message: Cannot read property 'length' of undefined
  Code: -32603 
```

It happened when I edited `self.window = ...` to read `let window = ...` in this code:

```swift
import UIKit

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {
    var window: UIWindow?

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
        let mainPage = MainPage()
        let navigator = UINavigationController(rootViewController: mainPage)

        self.window = UIWindow(frame: UIScreen.main.bounds)
        self.window!.backgroundColor = .white
        self.window!.rootViewController = navigator
        self.window!.makeKeyAndVisible()

        return true
    }
}
```

**Comment by vknabel** [comment]
@LinusU good catch! Was actually related to #11 and occured if there is no documentation like in your `window`.

[vscode-swift-development-environment-2.3.0.vsix.zip](https://github.com/vknabel/vscode-swift-development-environment/files/1822147/vscode-swift-development-environment-2.3.0.vsix.zip)

**Comment by vknabel** [comment]
Fixed in 2.3.0
